src/principal.rs

```diff
@@ -113,131 +84,155 @@ impl TryFrom<u8> for PrincipalClass {
 /// );
 /// ```
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct Principal(PrincipalInner);
+#[repr(packed)]
```

Comment:
Relating to what you brought up in the PR description, I think unless we can accompany this with some plausible description of "extreme requirements," we should remove `#[repr(packed)]` here.

This is from the [Rust Nomicon](https://doc.rust-lang.org/nomicon/other-reprs.html#reprpacked):
> repr(packed) is not to be used lightly. Unless you have extreme requirements, this should not be used.

Reply:
Removed after verification.